packages/manager/modules/vps/src/dashboard/vps-dashboard.routing.js

```diff
@@ -98,6 +102,79 @@ export default /* @ngInject */ ($stateProvider) => {
         $state.go(`vps.detail.dashboard.configuration.upgrade`, {
           upgradeType,
         }),
+
+      vpsMigration: /* @ngInject */ ($http, serviceName) =>
+        $http.get(`/vps/${serviceName}/migration2014`),
+
+      isMigrationRequired: /* @ngInject */ (vpsMigration) =>
+        get(vpsMigration, 'data.status') === 'toPlan',
+
+      goToVpsMigration: /* @ngInject */ (
+        $state,
+        vpsMigration,
+        catalog,
+        stateVps,
+        VpsMigrationService,
+      ) => () => {
+        const migrationPlan = VpsMigrationService.getMigrationPlan(
+          catalog,
+          vpsMigration.data.model,
+        );
+        const server = assign({
+          description: VpsMigrationService.getVpsModelDescription(
+            stateVps.model,
+          ),
+          migration: VpsMigrationService.populateOptionsPrice(
+            vpsMigration.data,
+            catalog,
+          ),
+          migrationPlan,
+          migrationDescription: VpsMigrationService.getVpsModelDescription(
+            migrationPlan.description,
+          ),
+        });
+        return $state.go(`vps.detail.dashboard.schedule`, {
+          servers: [server],
+        });
+      },
+    },
+  });
+  $stateProvider.state('vps.detail.dashboard.schedule', {
+    url: '/migration/schedule',
+    component: scheduleComponent.name,
+    translations: {
+      value: ['.'],
+      format: 'json',
+    },
+    layout: 'modal',
+    redirectTo: (transition) =>
+      transition
+        .injector()
+        .getAsync('servers')
+        .then((servers) =>
+          servers && servers.length > 0
+            ? false
+            : {
+                state: 'vps.migration',
+              },
+        ),
+    params: { servers: null },
+    resolve: {
+      servers: /* @ngInject */ ($transition$) => $transition$.params().servers,
+      goBack: /* @ngInject */ ($state, CucCloudMessage) => (
+        message = false,
+        type = 'success',
```

Comment:
"success"  should come from a constant

Reply:
Sure it can. But it is just used for `CucCloudMessage[type]`, and it is equivalent to calling CucCloudMessage.success. I guess it is not absolutely required to use a constant here